src/cpp/session/modules/SessionConnections.R

```diff
@@ -160,26 +160,107 @@ options(connectionObserver = list(
    NULL
 })
 
+.rs.addFunction("connectionReadSnippets", function() {
+   snippetsPath <- getOption("connections-path", "/etc/rstudio/connections/")
```

Comment:
I'd start with a global session option which defines connections available to everyone (this can be done in open source), and later add a Pro profile option which defines *additional* connections available to particular groups.